pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationDataSource.java

```diff
@@ -82,39 +83,77 @@ default String getHttpHeader(String name) {
 
     /**
      * Check if data from Pulsar protocol are available.
-     * 
+     *
      * @return true if this authentication data contain data from Pulsar protocol
      */
     default boolean hasDataFromCommand() {
         return false;
     }
 
     /**
-     * 
+     *
      * @return authentication data which is stored in a command
      */
     default String getCommandData() {
         return null;
     }
 
+
+    /*
+     * SASL
+     */
+
+    /**
+     *
+     * @return authentication data which will be stored in a command
```

Comment:
This is changing the current authentication framework in non-trivial ways. This of course is needed to support Kerberos but It needs to be done in a separated PR. Also we should have a PIP to outline the design and have more people to get eyes  on it.